### PR TITLE
change relative content paths to absolute

### PR DESCRIPTION
**Pull request by benq66**
I've tried to futureproof the function a bit, therefore it has some redundant checks since we only have images as content atm.

### CODE REVIEW

File: src/h5p/H5P.util.ts [hunk 1]
```diff
@@ -1,4 +1,44 @@
...
+ * @param contentId Content id of the H5P being shown
+ * @returns {void}
+ */
+export const addAbsolutePathToFiles = (
+  items: Array<TopicMapItemType> | undefined,
+  contentId: string,
+): void => {
+  if (!items) return;
+
+  items.map(item => {
```
Comment (boyum) [inline src/h5p/H5P.util.ts:21]: Remember to return the mapped array with the changes. `Array#map` will not change in place, but rather return an updated copy.

```suggestion
export const addAbsolutePathToFiles = (
  items: Array<TopicMapItemType> | undefined,
  contentId: string,
): Array<TopicMapItemType> | undefined => {
  if (!items) return;

  return items.map(item => {
```

File: src/h5p/H5PWrapper.tsx [hunk 2]
```diff
@@ -14,6 +14,8 @@ export class H5PWrapper extends H5P.EventDispatcher implements IH5PWrapper {
 
     console.info({ params, contentId, extras });
 
+    addAbsolutePathToFiles(params.topicMap.topicMapItems, contentId);
```
Comment (boyum) [inline src/h5p/H5PWrapper.tsx:17]:
```suggestion
    params.topicMap.topicMapItems = addAbsolutePathToFiles(params.topicMap.topicMapItems, contentId);
```

Comment (benq66) [reply src/h5p/H5PWrapper.tsx]: Updated! Thanks.

File: src/h5p/H5P.util.ts [hunk 3]
```diff
@@ -1,4 +1,44 @@
...
+  items: Array<TopicMapItemType> | undefined,
+  contentId: string,
+): Array<TopicMapItemType> | undefined => {
+  if (!items) return undefined;
+
+  return items.map(item => {
+    if (item.backgroundImage) {
+      const updatedItem = { ...item };
+
+      if (updatedItem.backgroundImage) {
```
Comment (boyum) [inline src/h5p/H5P.util.ts:25]: Testing for both `item.backgroundImage` and `updatedItem.backgroundImage` is redundant and strictly unecessary.

Comment (benq66) [reply src/h5p/H5P.util.ts:25]: I had it like this initially, but got an error 😢
Code_5XCWm3vaaB

File: src/h5p/H5P.util.ts [hunk 4]
```diff
@@ -1,4 +1,44 @@
...
+      // if (updatedItem.soundFile) {
+      //   ...
+      // }
+
+      return updatedItem;
+    }
+
+    return item;
+  });
+};
```
Comment (boyum) [inline src/h5p/H5P.util.ts:44]: Perhaps it's easier if we change this to only update the background image path:

```suggestion
export const makeBackgroundImagePathsAbsolute = (
  items: Array<TopicMapItemType> | undefined,
  contentId: string,
): Array<TopicMapItemType> | undefined => {
  if (!items) return undefined;

  return items.map(item => {
    if (!item.backgroundImage) {
      return item;
    }

    const pathAlreadyAbsolute =
      item.backgroundImage.path.startsWith("http://") ||
      item.backgroundImage.path.startsWith("https://");

    return { 
       ...item,
       backgroundImage: {
         ...item.backgroundImage,
         path: pathAlreadyAbsolute
           ? item.backgroundImage.path
           : H5P.getPath(item.backgroundImage.path, contentId);
       };
    };
  });
};
```

Comment (benq66) [reply src/h5p/H5P.util.ts]: Updated!

**Review by boyum — APPROVED** [review]

**Pull request merged**